src/api/Intelligence.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * @flow
+ * @file Helper for the Box AI API endpoint
+ * @author Box
+ */
+
+import Base from './Base';
+import { ERROR_CODE_INTELLIGENCE } from '../constants';
+import type { BoxItem } from '../common/types/core';
+import type { ElementsErrorCallback } from '../common/types/api';
+
+class Intelligence extends Base {
+    /**
+     * @property {Function}
+     */
+    successCallback: Function;
+
+    /**
+     * @property {Function}
+     */
+    errorCallback: ElementsErrorCallback;
+
+    /**
+     * API endpoint to ask ai a question
+     *
+     * @param {string} prompt - Question
+     * @param {Array<object>} items - Array of items to ask about
+     * @return {Promise}
+     */
+    async ask(prompt: string, items: Array<BoxItem>, successCallback: Function, errorCallback: ElementsErrorCallback) {
+        this.errorCode = ERROR_CODE_INTELLIGENCE;
```

Comment:
Why is this error code set here like this?

Reply:
this.errorCode is used by the base class to be applied when the errorHandler is being used